packages/webapp/pages/_app.tsx

```diff
@@ -92,14 +92,14 @@ function InternalApp({ Component, pageProps, router }: AppProps): ReactElement {
       return undefined;
     }
 
-    const onBeforeHistoryChange = () => {
+    const onRouteChange = () => {
       closeModal();
     };
 
-    router.events.on('beforeHistoryChange', onBeforeHistoryChange);
+    router.events.on('routeChangeStart', onRouteChange);
 
     return () => {
-      router.events.off('beforeHistoryChange', onBeforeHistoryChange);
+      router.events.off('routeChangeStart', onRouteChange);
```

Comment:
Move to `routeChangeStart` because `beforeHistoryChange` was triggered unintentionally on page load when calling `openModal`. Not sure exactly why. 

Also tested closing other modals that were mentioned in https://github.com/dailydotdev/apps/pull/1742 and they work.